platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -1056,10 +1056,20 @@ private NewCookie createCookie(
           url = new URL(request.getRequestURL().toString());
           LOGGER.debug("Returning new cookie for user.");
 
+          String servicesIdpPath = SystemBaseUrl.INTERNAL.getRootContext() + "/idp";
+
+          if (url.getHost().equals(SystemBaseUrl.EXTERNAL.getHost())
+              && String.valueOf(url.getPort()).equals(SystemBaseUrl.EXTERNAL.getPort())) {
+            servicesIdpPath =
+                SystemBaseUrl.EXTERNAL.getRootContext()
+                    + SystemBaseUrl.INTERNAL.getRootContext()
+                    + "/idp";
```

Comment:
❓ Why are we not doing `SystemBaseUrl.EXTERNAL.constructUrl()` like we are in other similar places?

Reply:
That will construct the entire URL. We just need the `/ddf/services/idp` part